src/System.Private.CoreLib/shared/System/TimeZoneInfo.cs

```diff
@@ -1989,5 +1979,88 @@ private static void ValidateTimeZoneInfo(string id, TimeSpan baseUtcOffset, Adju
                 }
             }
         }
+
+        private static readonly TimeSpan MaxOffset = TimeSpan.FromHours(14.0);
+        private static readonly TimeSpan MinOffset = -MaxOffset;
+        
+        /// <summary>
+        /// Helper function that validates the TimeSpan is within +/- 14.0 hours
+        /// </summary>
+        internal static bool UtcOffsetOutOfRange(TimeSpan offset) =>
+            offset < MinOffset || offset > MaxOffset;
+
+        private static TimeSpan GetUtcOffset(TimeSpan baseUtcOffset, AdjustmentRule adjustmentRule)
+        {
+            return baseUtcOffset
+                + adjustmentRule.BaseUtcOffsetDelta
+                + (adjustmentRule.HasDaylightSaving ? adjustmentRule.DaylightDelta : TimeSpan.Zero);
+        }
+        
+        /// <summary>
+        /// Helper function that performs adjustment rule validation
+        /// </summary>
+        private static bool IsValidAdjustmentRuleOffest(TimeSpan baseUtcOffset, AdjustmentRule adjustmentRule)
+        {
+            TimeSpan utcOffset = GetUtcOffset(baseUtcOffset, adjustmentRule);
+            return !UtcOffsetOutOfRange(utcOffset);
+        }
+
+        /// <summary>
+        /// Normalize adjustment rule offset so that it is within valid range
+        /// This method should not be called at all but is here in case something changes in the future
```

Comment:
>This method should not be called at all but is here in case somethin [](start = 12, length = 68)

this is confusing comment as we already calling this method.

Reply:
After reading the rest of the code, I don't think we need to have this method at all. and in the places we call it, we'll just assert there.

---
In reply to: [193198508](https://github.com/dotnet/coreclr/pull/18305#discussion_r193198508) [](ancestors = 193198508)